PWGCF/Tasks/correlations.cxx

```diff
@@ -751,21 +770,28 @@ struct CorrelationTask {
       LOGF(info, "processMCSameDerived. MC collision: %d, particles: %d, collisions: %d", mcCollision.globalIndex(), mcParticles.size(), collisions.size());
     }
 
-    same->fillEvent(mcCollision.multiplicity(), CorrelationContainer::kCFStepAll);
-    fillCorrelations<CorrelationContainer::kCFStepAll>(same, mcParticles, mcParticles, mcCollision.multiplicity(), mcCollision.posZ(), 0, 1.0f);
+    auto multiplicity = mcCollision.multiplicity();
+    if (cfgCentBinsForMC > 0) {
+      for (auto& collision : collisions) {
+        multiplicity = collision.multiplicity();
+      }
+    }
+
```

Comment:
This is not ideal. In the case that there is no collision at all, we fill with `mcCollision.multiplicity()`
Better: In case `cfgCentBinsForMC > 0` and `collisions.size() == 0)` do a `return`